dotCMS/src/integration-test/java/com/dotcms/publishing/PublisherAPITest.java

```diff
@@ -319,4 +327,304 @@ private FileFilter getNoBundleXMLFileFilter() {
+     */
+    @Test
+    public void test_getFilterDescriptorByKey_filterKeyDoesNotExist_returnDefaultFilter(){
+        PublisherAPIImpl.class.cast(publisherAPI).getFilterDescriptorMap().clear();
+
+        final String keyDefault = "TestByKeyDoesNotExistDefault.yml";
+        createFilterDescriptor(keyDefault,"TestByKeyDoesNotExistDefault",true,null);
+
+        final String keyNonDefault = "TestByKeyDoesNotExistNonDefault.yml";
+        createFilterDescriptor(keyNonDefault,"TestByKeyDoesNotExistNonDefault",false,null);
+
+        final FilterDescriptor filterDescriptor = publisherAPI.getFilterDescriptorByKey("thisKeyNotExists");
+        Assert.assertNotNull(filterDescriptor);
+        Assert.assertEquals(keyDefault,filterDescriptor.getKey());
+    }
+
+    /**
+     * This test creates a FilterDescriptor and a Bundle (with the created FilterDescriptor) and
+     * returns a PublisherFilter with the filters of the FilterDescriptor.
+     *
+     * In this case the FilterDescriptor has all the possible filters.
+     */
+    @Test
+    public void test_createPublisherFilter_withAllFilters()
+            throws DotDataException, DotSecurityException {
+        PublisherAPIImpl.class.cast(publisherAPI).getFilterDescriptorMap().clear();
+
+        final String filterKey = "TestCreatePublisherFilterAllFilters.yml";
+        final Map<String,Object> filtersMap = new HashMap<>();
+        filtersMap.put("excludeQuery","+baseType:5");
+        final List<Object> listExcludeClasses = new ArrayList();
+        listExcludeClasses.add("User");
+        listExcludeClasses.add("Host");
+        listExcludeClasses.add("ContentType");
+        filtersMap.put("excludeClasses", listExcludeClasses);
+        filtersMap.put("dependencies",true);
+        filtersMap.put("excludeDependencyQuery","+baseType:7");
+        filtersMap.put("excludeDependencyClasses", listExcludeClasses);
+        filtersMap.put("forcePush",false);
+        filtersMap.put("relationships",true);
+        createFilterDescriptor(filterKey,"TestCreatePublisherFilter",true,filtersMap);
+
+        final String bundleName = "testCreatePublisherFilterAllFilters";
+        Bundle bundle = new Bundle(bundleName, new Date(), null, adminUser.getUserId(),
+                (Boolean) filtersMap.get("forcePush"),filterKey);
+        APILocator.getBundleAPI().saveBundle(bundle);
+        bundle = APILocator.getBundleAPI().getBundleByName(bundleName);
+        Assert.assertNotNull(bundle);
+        Assert.assertEquals(filterKey,bundle.getFilterKey());
+
+        final PublisherFilter publisherFilter = publisherAPI.createPublisherFilter(bundle.getId());
+        Assert.assertNotNull(publisherFilter);
+        Assert.assertEquals(filtersMap.get("dependencies"),publisherFilter.isDependencies());
+        Assert.assertEquals(filtersMap.get("relationships"),publisherFilter.isRelationships());
+        Assert.assertEquals(filtersMap.get("forcePush"),bundle.isForcePush());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of fully qualified name 'Assert.assertEquals' due to existing static import 'org.junit.Assert.assertEquals'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5458951)